[Soumyadip-cmd] ip

Soumyadip-cmd opened the pull request:
### Assistinator
> “If you spend too much time thinking about a thing, you’ll never get it done.” – Bruce Lee ([source](https://dansilvestre.com/productivity-quotes/))

Assistinator helps you reduce time thinking about what you have to do and lets get started on the things you need to do. It's

- text-based
- ~~easy~~ super easy to use
- fast

All you need to do is,

1. download it from [here](https://github.com/Soumyadip-cmd/ip/releases)
2. double click it
3. add your tasks 
4. let it manage your tasks for you :relaxed:

And it is **Free**!

Features:

- [x] Managing tasks
- [ ] Managing deadlines
- [ ] Reminders (Coming soon)
___________________________________________________________
If you are a Java programmer, you can use it to practice Java too. Here is the `main` method:
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Soumyadip-cmd committed:
Rename, Greet, Exit

Soumyadip-cmd committed:
Echo

Soumyadip-cmd committed:
Add, List

Soumyadip-cmd committed:
Mark as Done

Soumyadip-cmd committed:
ToDos, Events, Deadlines

Soumyadip-cmd committed:
Automated Text UI Testing

Soumyadip-cmd committed:
Handle Errors

Soumyadip-cmd committed:
Delete

Soumyadip-cmd committed:
Use Enums

Soumyadip-cmd committed:
implement saving tasks in the hard disk

Soumyadip-cmd committed:
Merge branch 'branch-Level-7'

Soumyadip-cmd committed:
Add dates in deadline

Soumyadip-cmd committed:
Merge branch 'branch-Level-8'

Soumyadip-cmd committed:
add more OOP

Soumyadip-cmd committed:
Merge branch 'More-oop'

Soumyadip-cmd committed:
Organise into package

Soumyadip-cmd committed:
Merge branch 'add-gradle-support'

Soumyadip-cmd committed:
Add JUnit tests for storage and parser classes

Soumyadip-cmd committed:
Merge branch 'JUnit-tests'

Soumyadip-cmd committed:
Create a JAR file

Soumyadip-cmd committed:
Add Javadoc comments

Soumyadip-cmd committed:
Refactor code to follow coding convention

Soumyadip-cmd committed:
add find command and functions

Soumyadip-cmd committed:
Merge branch 'add-javadoc-comments'

Soumyadip-cmd committed:
Fix merge conflict when merging add-find-functionality

Soumyadip-cmd committed:
Merge branch 'refactor-code-to-follow-convention'

jan-kai1 commented on src/main/java/Assistinator/AssitinatorExceptions.java:
```diff
@@ -0,0 +1,7 @@
+package Assistinator;
+
+public class AssitinatorExceptions extends Exception{
```
Typo with Assitnator instead of Assistinator, might want to consider changing class to AssistinatorException instead of Exceptions as there is only 1 exception

jan-kai1 commented on src/main/java/Assistinator/Command.java:
```diff
@@ -0,0 +1,14 @@
+public enum Command {
+    BYE, LIST, MARK, UNMARK, TODO, EVENT, DEADLINE, DELETE, FIND, UNKNOWN;
+
+    static Command fromString(String input) {
```
Might consider naming the method `parse` or `parseCommand` something similar instead of fromString to be more descriptive

jan-kai1 commented on src/main/java/Assistinator/Deadline.java:
```diff
@@ -0,0 +1,35 @@
+import java.time.format.DateTimeFormatter;
+
+public class Deadline extends Task {
+    protected LocalDate by;
```
may consider changing the by to be a final variable as it is not modified

jan-kai1 commented on src/main/java/Assistinator/Deadline.java:
```diff
@@ -0,0 +1,35 @@
+     * Generate string to print onto file
+     * @return String for file
+     */
+    public String toFileString() {
```
Consider naming `toStorageString` instead of `toFileString` as could be confused for file path string

jan-kai1 commented on src/main/java/Assistinator/Deadline.java:
```diff
@@ -0,0 +1,35 @@
+    /**
+     * Initialise deadline task
+     * @param description task description
+     * @param by deadline
```
Could specify more on the param by (the format of the string argument)

jan-kai1 commented on src/main/java/Assistinator/Parser.java:
```diff
@@ -0,0 +1,80 @@
+    private Event parseEvent(String input) throws AssitinatorExceptions {
+        String[] parts = input.split(" /");
+        if (parts.length != 3 || parts[1].equals("from") || parts[2].equals("to")) {
+            throw new AssitinatorExceptions("Please follow format: event {task description} /from {start} /to {end}");
```
i like the different error messages for each different task type

jan-kai1 reviewed (commented):
LGTM, minor issues with method naming for clarity

Soumyadip-cmd committed:
Add checkstyle

Soumyadip-cmd committed:
Merge branch 'add-checkstyle'

Soumyadip-cmd committed:
Add JavaFX GUI

Soumyadip-cmd committed:
Merge branch 'branch-Level-10'

sannie-beep commented on src/main/java/assistinator/Deadline.java:
```diff
@@ -0,0 +1,38 @@
+     */
+    @Override
+    public String toString() {
+        return "[D]" + super.toString() + " (by: " + by.format(DateTimeFormatter.ofPattern("MMM d yyyy")) + ")";
```
Perhaps the date time format could be a constant in the superclass Task such that the Event class could also inherit it?

sannie-beep commented on src/main/java/assistinator/Parser.java:
```diff
@@ -0,0 +1,83 @@
+        return new Deadline(description, deadline);
+    }
+
+    private Event parseEvent(String input) throws AssitinatorExceptions {
```
Perhaps implementing Date time formatting for Events and checking user input could make the formatting more standardized. Nice work with organizing and clarifying your implementation using Javadocs though! :)

sannie-beep commented on src/test/java/assistinator/StorageTest.java:
```diff
@@ -0,0 +1,34 @@
+
+import org.junit.jupiter.api.Test;
+
+public class StorageTest {
```
Nice naming of the tests to explicate exactly what they are for! It would be nice to include javadocs to make the tests understandable to contributors. But overall, this is well-formatted and suits the coding conventions! 👍

sannie-beep reviewed (commented):
Overall,  code follows the coding standard quite well. Perhaps consider adding javadocs in the testing, and perhaps use a constant for the date time formatting. Good job so far! :))

Soumyadip-cmd committed:
Add assert statements

Soumyadip-cmd committed:
Improve code quality

Assistinator class handles execution of command. 0 and 1 are used to
represent if a task is not done or done respectively.

Multiple levels of abstraction in Assistinator class leading to
violation of SLAP. Arbitrary numbers 0 and 1 used to represent state of
tasks makes code not obvious.

Let's,
* extract all functions to execute command into a separate class
* create a enum class to represent states of tasks

CommandExecutor class helps to abstract away command execution logic so
that Assistinator focuses on higher level application logic. Using DONE
NOTDONE for task status is more intuitive.

Soumyadip-cmd committed:
Add assert check in parser.parseTask

Soumyadip-cmd committed:
Refactor task methods to use Java streams

Soumyadip-cmd committed:
Merge pull request #1 from Soumyadip-cmd/branch-A-CodeQuality

Improve code quality

Soumyadip-cmd committed:
Merge branch-A-Assertions to master

Soumyadip-cmd committed:
Merge branch-A-Streams to master

Soumyadip-cmd committed:
Revert "Merge branch-A-Streams to master"

This reverts commit 26be83e26d2b3b8c2217b0b01f7b77907e9faf55, reversing
changes made to ef73cfe8f842c21165252392c2b4c7eb196c342d.

Soumyadip-cmd committed:
Revert "Merge branch-A-Assertions to master"

This reverts commit ef73cfe8f842c21165252392c2b4c7eb196c342d, reversing
changes made to 49b28b2da7aefd7b38386f9b884c2eb2376fd156.

Soumyadip-cmd committed:
Merge branch 'master' into branch-A-Assertions

Soumyadip-cmd committed:
Merge branch 'master' into branch-A-Streams

Soumyadip-cmd committed:
Merge pull request #2 from Soumyadip-cmd/branch-A-Assertions

Branch a assertions

Soumyadip-cmd committed:
Merge branch 'master' into branch-A-Streams

Soumyadip-cmd committed:
Merge pull request #3 from Soumyadip-cmd/branch-A-Streams

Refactor task methods to use Java streams

Soumyadip-cmd committed:
Revert "Merge branch 'master' into branch-A-Assertions"

This reverts commit ab4429827967c406bb1c37760979b21c6dfb427b, reversing
changes made to 84c886ad3a86f275f66ed1f40ea3c34a30adeeb9.

Soumyadip-cmd committed:
Reapply "Merge branch-A-Streams to master"

This reverts commit 993be5f10547858cd2cd0be5f5f11078fd3d0d01.

Soumyadip-cmd committed:
Revert "Merge branch-A-Streams to master"

This reverts commit 26be83e26d2b3b8c2217b0b01f7b77907e9faf55, reversing
changes made to ef73cfe8f842c21165252392c2b4c7eb196c342d.

Soumyadip-cmd committed:
Merge branch 'branch-A-Assertions' of https://github.com/Soumyadip-cmd/ip

Soumyadip-cmd committed:
Revert "Merge branch 'branch-A-Assertions' of https://github.com/Soumyadip-cmd/ip"

This reverts commit 43dd69bcc774c26e0b62221d40b49da61dd2404a, reversing
changes made to fad7f1c7f987e696b5c576ed1809d3cc3d5d5668.

Soumyadip-cmd committed:
Add assert statements

Soumyadip-cmd committed:
Merge pull request #4 from Soumyadip-cmd/branch-A-Assertions

Branch a assertions

Soumyadip-cmd committed:
Refactor tasks methods to use java streams

Soumyadip-cmd committed:
Merge branch 'master' into branch-A-Streams

Soumyadip-cmd committed:
Merge pull request #5 from Soumyadip-cmd/branch-A-Streams

Refactor tasks methods to use java streams

Soumyadip-cmd committed:
Event tasks: add checking of schedule clashes

Users can add events even if there are clashes in time periods with
other existing events.

Users will not be aware of clashing events and will have to deal with
conflicting events later on.

Let's check for conflicts when adding new events with existing events.
If there are conflicts, warn the users which existing event conflicts
with the new event.

This informs users of clashes in schedule and the user can then take
actions to resolve the clashes.

Soumyadip-cmd committed:
Merge pull request #6 from Soumyadip-cmd/add-checking-for-clashes

Event tasks: add checking of schedule clashes

Soumyadip-cmd committed:
Reduce writes to database

Soumyadip-cmd committed:
Improve the GUI of app

Soumyadip-cmd committed:
Implement unique personality

Soumyadip-cmd committed:
Set title and app icon

Soumyadip-cmd committed:
Fix formatting of Javadoc header

Soumyadip-cmd committed:
Add error handling for missing data file

Soumyadip-cmd committed:
Add User Guide to Assistinator

Soumyadip-cmd committed:
Add UI image

Soumyadip-cmd committed:
Refactor commands into commands package

Soumyadip-cmd committed:
Create _config.yml

Soumyadip-cmd committed:
Update _config.yml

Soumyadip-cmd committed:
Fix formatting of alert in pages

Soumyadip-cmd committed:
Merge branch 'master' from origin

Soumyadip-cmd committed:
Fix bug in documentation

Soumyadip-cmd committed:
Fix command summary table

Soumyadip-cmd committed:
Add alerts to documentation

Soumyadip-cmd committed:
Add alerts to documentation

Soumyadip-cmd committed:
Add alerts to documentation

Soumyadip-cmd committed:
Update project documentation

Soumyadip-cmd committed:
Update project documentation

Soumyadip-cmd committed:
Update format of UG
